### PR TITLE
Log policy extra headers

### LINKED ISSUE #5782: Include `WWW-Authenticate`, `x-ms-version`, `x-ms-date` to the default list of non-redacted headers in the log sanitizer
**Is your feature request related to a problem? Please describe.**

I am debugging REST requests to a Storage Account hitting auth failures in DuckDB and the logs redacted headers like `WWW-Authenticate`, `x-ms-version`, `x-ms-date` which could have helped root-causing the issue.

```
[2024-07-11T12:09:11.2782773Z T: 23464] INFO  : HTTP Request : HEAD https://azduckdb.blob.core.windows.net/data/HuggingFaceFW/fineweb/CC-MAIN-2013-20/000_00000.parquet
authorization : REDACTED
user-agent : azsdk-cpp-storage-blobs/12.10.0 (Windows 10 Enterprise 6.3 22631 22621.1.amd64fre.ni_release.220506-1250)
x-ms-client-request-id : ebac0027-0250-42aa-87dc-91b1713ab0ef
x-ms-date : REDACTED
x-ms-version : REDACTED

[2024-07-11T12:09:12.2519756Z T: 23464] INFO  : HTTP Response (972ms) : 401 Server failed to authenticate the request. Please refer to the information in the www-authenticate header.
date : Thu, 11 Jul 2024 12:09:10 GMT
server : Microsoft-HTTPAPI/2.0
transfer-encoding : chunked
www-authenticate : REDACTED
x-ms-error-code : REDACTED
x-ms-request-id : b596ec8d-701e-0010-2a8b-d36a02000000
```

Including `x-ms-error-code` might also be helpful but I could not find documentation on this header to understand what it can contain.

**Describe the solution you'd like**

Include `WWW-Authenticate`, `x-ms-version`, `x-ms-date` in [`g_defaultAllowedHttpHeaders`](https://github.com/Azure/azure-sdk-for-cpp/blob/b1207fc87aa8fc5c82fc657c1228a8205eb19091/sdk/core/azure-core/src/http/log_policy.cpp#L73)

If this sounds OK I can likely contribute a PR.

**Describe alternatives you've considered**

It looks like DuckDB could customize redacted headers but doing that in the Azure SDK would be a more generic solution.

**Additional context**

Related issue: #3708 

`WWW-Authenticate` is included in .NET and Java SDKs here:
- .NET: https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/src/DiagnosticsOptions.cs#L44
- Java: https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/core/azure-core/src/main/java/com/azure/core/http/policy/HttpLogOptions.java#L69

`x-ms-version` and `x-ms-date` are *not* included there but the `Date` header is and the `api-version` query parameter also is, so that seems like a reasonable inclusion.

**Information Checklist**

Kindly make sure that you have added all the following information above and checkoff the required fields otherwise we will treat the issuer as an incomplete report
- [x] Description Added
- [x] Expected solution specified

### PR DESCRIPTION
**Pull request by gearama**
#closes #5782
Per customer request adding a few extra headers in the log 
# Pull Request Checklist

Please leverage this checklist as a reminder to address commonly occurring feedback when submitting a pull request to make sure your PR can be reviewed quickly:

See the detailed list in the [contributing guide](https://github.com/Azure/azure-sdk-for-cpp/blob/main/CONTRIBUTING.md#pull-requests).

- [ ] [C++ Guidelines](https://azure.github.io/azure-sdk/cpp_introduction.html)
- [ ] Doxygen docs
- [ ] Unit tests
- [ ] No unwanted commits/changes
- [ ] Descriptive title/description
  - [ ] PR is single purpose
  - [ ] Related issue listed
- [ ] Comments in source
- [ ] No typos
- [ ] Update changelog
- [ ] Not work-in-progress
- [ ] External references or docs updated
- [ ] Self review of PR done
- [ ] Any breaking changes?

### CODE REVIEW

**Review by antkmsft — CHANGES_REQUESTED** [review]
Is `WWW-Authenticate` safe to log by default? Do other SDKs log it? What has motivated this change, can you give more context? Should this be in the global list, or should some client library (Tables, for example) use this list only for its client? Does this PR syncs this list with some other language SDK? Where did you get these new headers from, why these headers?
Also, probably not the change to be merged before today's release is out.
Would customer be unblocked if they'd add these headers to the client options (`options.Log.AllowedHttpHeaders.insert()`), maybe they didn't know they could do that?

**Comment by gearama** [comment]
> Is `WWW-Authenticate` safe to log by default? Do other SDKs log it? What has motivated this change, can you give more context? Should this be in the global list, or should some client library (Tables, for example) use this list only for its client? Does this PR syncs this list with some other language SDK? Where did you get these new headers from, why these headers? Also, probably not the change to be merged before today's release is out. Would customer be unblocked if they'd add these headers to the client options (`options.Log.AllowedHttpHeaders.insert()`), maybe they didn't know they could do that?

This change si by customer request , .Net and Java log the same header :
WWW-Authenticate is included in .NET and Java SDKs here:

.NET: https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/src/DiagnosticsOptions.cs#L44
Java: https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/core/azure-core/src/main/java/com/azure/core/http/policy/HttpLogOptions.java#L69

**Comment by antkmsft** [comment]
Sorry, it looks like I freaked out for no reason, and it is my fault for not reading the linked item. I think the change is likely ok, just let me to re-review it later, after we cut today's Core release, it will make it to the next one. Especially if we can double-confirm with the identity v-team that "WWW-Authenticate" is ok to log by default.

**Comment by gearama** [comment]
> WWW-Authenti
Based on Conversation Ahson had with the Identity SDK v-team it is ok to lok this header as it is sent to unauthorized clients. 
https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/WWW-Authenticate

**Review by RickWinter — COMMENTED** [review]
Please add a CHANGELOG entry after we cut this months release.

Looks good other than that.

**Review by LarryOsterman — APPROVED** [review]

**Review by antkmsft — APPROVED** [review]

**Comment by gearama** [comment]
> Please add a CHANGELOG entry after we cut this months release.
> 
> Looks good other than that.

True that, i'll do it Fryday to give time for the release

**Review by ahsonkhan — APPROVED** [review]

File: sdk/core/azure-core/CHANGELOG.md [hunk 1]
```diff
@@ -3,7 +3,7 @@
 ## 1.14.0-beta.2 (Unreleased)
 
 ### Features Added
-
+- Request logs to now include the `accept-range`, `content-range`,`range`,`WWW-Authenticate`, `xx-ms-date`,`x-ms-error-code`,`x-ms-range`  and `x-ms-version` headers.
```
Comment (ahsonkhan) [inline sdk/core/azure-core/CHANGELOG.md:6]: consistency nit: add a new line between heading and text (vscode markdown linter also complains about this)

```suggestion

- Request logs to now include the `accept-range`, `content-range`,`range`,`WWW-Authenticate`, `xx-ms-date`,`x-ms-error-code`,`x-ms-range`  and `x-ms-version` headers.
```

**Pull request merged**